### DISCUSSION TITLE
ArduinoBLE vs NimBLE-Arduino

### DISCUSSION THREAD

**Discussion by antoniomolram** [h2zero/NimBLE-Arduino#454] [General]
Hi,

I am trying to get a better understanding of the different BLE arduino libraries. I see that one of the most loved ones is NimBLE-Arduino (number of starts on github), but Arduino has created ArduinoBLE.

Both support ESP32, and Nordic (Nicla sense for example is supported on arduinoBLE) chips, right? What are for you the main differences? 

Thank you!

**Comment by h2zero** [comment]
Great question, I see that ArduinoBLE just added esp32 support recently. I'm not familiar with it so I can't really say much about it.

I'd be interested to see responses from users that may have some experience with both libraries.
